### PR TITLE
Interceptor

### PR DESCRIPTION
**Pull request by amontenegro**

### CODE REVIEW

**Comment by amontenegro** [comment]
[Add "; charset=utf8" to content type in http headers to prevent ISO text being posted to the server](https://trello.com/c/LZVOoi2c/8801-add-charsetutf8-to-content-type-in-http-headers-to-prevent-iso-text-being-posted-to-the-server)

**Comment by amontenegro** [comment]
[Add "; charset=utf8" to content type in http headers to prevent ISO text being posted to the server](https://trello.com/c/LZVOoi2c/8801-add-charsetutf8-to-content-type-in-http-headers-to-prevent-iso-text-being-posted-to-the-server)

File: src/app/core/http-content-type-header-interceptor/http-content-type-header-interceptor.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+export class HttpContentTypeHeaderInterceptor implements HttpInterceptor {
+    construct () {
+
+    }
+
+    intercept(req:HttpRequest<any>, next: HttpHandler) {
+        var method = req.method
+        var urlWithParams = req.urlWithParams
+        // If the request contains a content type, be sure to set the encoding to utf-8        
+        if(['POST', 'PUT'].includes(method)) {
```
Comment (leomendoza123) [inline src/app/core/http-content-type-header-interceptor/http-content-type-header-interceptor.ts:14]: We might need to add `PATCH` in the future. 
NO a change we need right now

File: src/app/core/http-content-type-header-interceptor/http-content-type-header-interceptor.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            var clonedRequest;
+            if(urlWithParams == '/signin/auth.json') {
```
Comment (leomendoza123) [inline src/app/core/http-content-type-header-interceptor/http-content-type-header-interceptor.ts:16]: this header excepcion seems to be require for `userStatus.json?logUserOut=true, sendReactivation.json, social/signin/auth.json, shibboleth/signin/auth.json, signin/auth.json`


Maybe we can declare a class level atribute like

```
const formUrlEcondedUrls = ["userStatus.json?logUserOut=true", "sendReactivation.json", "social/signin/auth.json", "shibboleth/signin/auth.json", "signin/auth.json"];
```

And check if a route is in the array 
``` 
if(formUrlEcondedUrls.includes(urlWithParams)) {
```

**Review by leomendoza123 — APPROVED** [review]

**Pull request merged**